### PR TITLE
Asish.ap/integration testing

### PR DESCRIPTION
**Pull request by LijuJoseJJ**
# Description

Implemented integration test.

## Motivation and Context

Integration test need to setup Travis CI.

## How Has This Been Tested

By running new bash script.
```sh
./zkp-demo-test
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: API-Gateway/package.json [hunk 1]
```diff
@@ -5,8 +5,8 @@
   "repository": "https://github.com/EYBlockchain/nightfall",
   "main": "index.js",
   "scripts": {
-    "start": "NODE_ENV=LOCAL nodemon --exec babel-node ./src/index.js",
-    "test": "./test-run",
+    "start": "nodemon --exec babel-node ./src/index.js",
+    "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (ChaitanyaKonda) [inline API-Gateway/package.json]: Don't we have a unit test here for API Gateway?

File: integration-test/testData.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		email: "bob@ey.com",
+		password: "pass",
+	},
+	erc721: {
+		tokenURI: "one",
+		tokenID: generateTokenID(),
+	},
+	erc20: {
+		mint: 5,
+		tobeShield: [2, 3],
```
Comment (ChaitanyaKonda) [inline integration-test/testData.js]: Use consistent language with the rest of the code, documentation and paper. 
`toBeMintedAsCommitment` and not `toBeShield`.

**Comment by ChaitanyaKonda** [comment]
Let us increase the timeout time in integration-tests to make room for low config machines.

File: API-Gateway/package.json (hunk 1 above)
Comment (LijuJoseJJ) [reply API-Gateway/package.json]: currently we don't. 
unit test for api-gateway means to mock call to other service, and solely test api-gateway orchestration layer. right ?

File: integration-test/testData.js (hunk 2 above)
Comment (LijuJoseJJ) [reply integration-test/testData.js]: done

**Comment by LijuJoseJJ** [comment]
> Let us increase the timeout time in integration-tests to make room for low config machines.

done

**Comment by ocgordon** [comment]
@LijuJoseJJ Changed the `./zke-demo-test` to `./zkp-demo-test`to reduce confusion for testing.

**Comment by LijuJoseJJ** [comment]
> I ran this and got 18 passing tests, but encountered an error at the end of the run. Is this leftover code from something that was removed?
> 
> ```
> *** Remove test-net network if not removed ***
> Error: No such network: nightfall_test_net
> tab 1 of window id 138870
> ```

@ocgordon 
this is fail safe code.
explicitly removing network if docker-compose down fails to do so.

**Comment by ocgordon** [comment]
> > I ran this and got 18 passing tests, but encountered an error at the end of the run. Is this leftover code from something that was removed?
> > ```
> > *** Remove test-net network if not removed ***
> > Error: No such network: nightfall_test_net
> > tab 1 of window id 138870
> > ```
> 
> @ocgordon
> this is fail safe code.
> explicitly removing network if docker-compose down fails to do so.

Can we change the error message to make it explicit that this is intentional?

**Comment by LijuJoseJJ** [comment]
@ocgordon 


> > > I ran this and got 18 passing tests, but encountered an error at the end of the run. Is this leftover code from something that was removed?
> > > ```
> > > *** Remove test-net network if not removed ***
> > > Error: No such network: nightfall_test_net
> > > tab 1 of window id 138870
> > > ```
> > 
> > 
> > @ocgordon
> > this is fail safe code.
> > explicitly removing network if docker-compose down fails to do so.
> 
> Can we change the error message to make it explicit that this is intentional?

@ocgordon 

now this line code will only run if network is not removed by ```docker-compose . down```.

**Pull request merged**